Library/Homebrew/cmd/upgrade.rb

```diff
@@ -41,7 +42,10 @@ def upgrade
       puts pinned.map { |f| "#{f.full_name} #{f.pkg_version}" } * ", "
     end
 
-    outdated.each { |f| upgrade_formula(f) }
+    outdated.each do |f|
+      upgrade_formula(f)
```

Comment:
I don't see any benefit of adding the clause `&& f.installed?`, as a [check](https://github.com/Homebrew/homebrew/blob/88ed0ee3f6f903bde4ddd674196e2a9afdc8019f/Library/Homebrew/cmd/upgrade.rb#L17-L22) is done prior to calling `upgrade_formula`. So, `if ARGV.include?("--cleanup") && f.installed?` would always be true.